src/torchcodec/_core/Frame.cpp

```diff
@@ -8,6 +8,11 @@
 
 namespace facebook::torchcodec {
 
+FrameDims::FrameDims(int height, int width) : height(height), width(width) {
+  TORCH_CHECK(height > 0, "FrameDims.height must be > 0, got: ", height);
+  TORCH_CHECK(width > 0, "FrameDims.width must be > 0, got: ", width);
+}
```

Comment:
Drive-by: when passing in a height and width, we should only be able to instantiate a `FrameDims` object with positive values. If we want a `FrameDims` object that has 0 for both values, that's just the default constructor. We should never have a `FrameDims` object with negative values.